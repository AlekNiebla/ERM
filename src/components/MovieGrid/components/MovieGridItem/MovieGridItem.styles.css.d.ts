export const main: string;
export const poster: string;
export const noPoster: string;
export const data: string;
export const title: string;
export const releaseDate: string;
export const genres: string;
