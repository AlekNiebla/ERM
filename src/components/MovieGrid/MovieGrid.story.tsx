import * as React from "react";
import { storiesOf } from "@storybook/react";
// component / interface
import MovieGrid, { IMovie } from "./MovieGrid";

const movieList: IMovie[] = [
  {
    id: 1,
    title: "Pulp Fiction",
    poster: "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
    releaseDate: "1994-09-10",
    genres: ["Thriller", "Crime"]
  },
  {
    id: 2,
    title: "No poster prop",
    releaseDate: "2019-09-10",
    genres: ["Crime"]
  },
  {
    id: 3,
    title: "Empty poster string",
    poster: "",
    releaseDate: "1994-09-10",
    genres: ["Thriller", "Crime"]
  },
  {
    id: 4,
    title: "Wrong date format",
    releaseDate: "19"
  }
];

storiesOf("Components / MovieGrid", module)
  .add("default", () => <MovieGrid movies={movieList} />)
  .add("no movies", () => <MovieGrid />);
