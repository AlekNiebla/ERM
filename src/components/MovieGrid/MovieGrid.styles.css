.main {
}
.results {
}
.count {
}
.sorting {
}
.grid {
}
.sorryMessage {
}
