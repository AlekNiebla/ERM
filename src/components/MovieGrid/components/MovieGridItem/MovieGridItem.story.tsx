import * as React from "react";
import { storiesOf } from "@storybook/react";
import MovieGridItem, { IMovieGridItem } from "./MovieGridItem";
// mock data
const singleMovie: IMovieGridItem = {
  id: 1,
  title: "Pulp Fiction",
  poster: "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
  releaseDate: "1994-09-10",
  genres: ["Thriller", "Crime"]
};

storiesOf("Components / MovieGrid / MovieGridItem", module).add(
  "default",
  () => (
    <div
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "#f1f2f3",
        padding: "100px"
      }}
    >
      <MovieGridItem {...singleMovie} />
    </div>
  )
);
