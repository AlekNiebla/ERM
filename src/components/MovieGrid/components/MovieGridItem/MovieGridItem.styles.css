.main {
  border-radius: var(--step-3);
  overflow: hidden;
  max-width: 250px;
  border: 1px solid var(--color-gray-lighter);
  background-color: var(--color-white);
  box-shadow: 0 2px 5px 0 var(--color-gray-lighter);
  position: relative;
}
.poster {
  max-width: 100%;
  display: block;
  z-index: 1;
}
.noPoster {
  width: 100%;
  display: block;
  height: 100%;
  color: var(--color-gray-dark);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1;
}
.data {
  display: flex;
  flex-direction: column;
  padding: var(--step-3);
  position: absolute;
  bottom: 0;
  width: 100%;
  z-index: 4;
  background-color: var(--color-white-transparent);
}
.baseData {
  margin-bottom: var(--step-2);
}
.title {
  composes: Typo-Big-Bold from global;
  color: var(--color-gray-darkest);
  margin-right: var(--step-2);
}
.releaseDate {
  composes: Typo-Base from global;
  color: var(--color-primary);
}
.genres {
  composes: Typo-Small from global;
  color: var(--color-gray-dark);
}
