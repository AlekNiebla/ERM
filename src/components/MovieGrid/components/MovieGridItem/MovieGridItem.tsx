import * as React from "react";

const MovieGridItem = () => <span>MovieGridItem</span>;

export default MovieGridItem;
