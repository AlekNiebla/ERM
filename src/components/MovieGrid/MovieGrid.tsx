import * as React from "react";
// components
import MovieGridItem from "./components/MovieGridItem/MovieGridItem";
// interface
interface IMoviesGird {
  similarResults: false;
  movies?: any[];
}

const MovieGrid = (props: IMoviesGird) => (
  <div className="main">
    {props.similarResults ? (
      <div>
        <span>Films by </span>
        <span>Drama genre</span>
      </div>
    ) : (
      <div className="results">
        <span className="count">7 movies found</span>
        <div className="sorting">
          <span>Sort by</span>
          <button>release date</button>
          <button>rating</button>
        </div>
      </div>
    )}
    {props.movies ? (
      <div className="grid">
        <MovieGridItem />
        <MovieGridItem />
        <MovieGridItem />
        <MovieGridItem />
        <MovieGridItem />
        <MovieGridItem />
        <MovieGridItem />
      </div>
    ) : (
      <div>
        <span>No movies :(</span>
      </div>
    )}
  </div>
);

export default MovieGrid;
