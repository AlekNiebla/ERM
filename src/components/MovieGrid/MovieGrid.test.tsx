import * as React from "react";
import { shallow } from "enzyme";
// components
import MovieGrid, { IMovie } from "./MovieGrid";
import MovieGridItem from "./components/MovieGridItem/MovieGridItem";
// styles
import * as styles from "./MovieGrid.styles.css";

const mockMovie: IMovie = {
  id: 1,
  title: "Pulp Fiction",
  poster: "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
  releaseDate: "1994-09-10",
  genres: ["Thriller", "Crime"]
};

describe("MovieGrid component...", () => {
  it("... renders without errors", () => {
    const wrapper = shallow(<MovieGrid />);
    expect(wrapper.find(MovieGrid)).toBeTruthy();
  });
  it("... matches snapshot", () => {
    const wrapper = shallow(<MovieGrid />);
    expect(wrapper).toMatchSnapshot();
  });
  it("... renders sorting options", () => {
    const wrapper = shallow(<MovieGrid />);

    expect(wrapper.find(`.${styles.sorting}`)).toBeTruthy();
    expect(wrapper.find(<button>release date</button>)).toBeTruthy();
    expect(wrapper.find(<button>rating</button>)).toBeTruthy();
  });
  it("... renders similar results category", () => {
    const wrapper = shallow(<MovieGrid similarResults={true} />);
    expect(wrapper.find(<span>Films by </span>)).toBeTruthy();
  });
  it("... renders movies", () => {
    const wrapper = shallow(<MovieGrid movies={[mockMovie]} />);
    expect(wrapper.find(`.${styles.grid}`)).toBeTruthy();
    expect(wrapper.find(MovieGridItem)).toBeTruthy();
  });
  it("... renders error message", () => {
    const wrapper = shallow(<MovieGrid />);
    expect(wrapper.find(`.${styles.sorryMessage}`)).toBeTruthy();
  });
});
