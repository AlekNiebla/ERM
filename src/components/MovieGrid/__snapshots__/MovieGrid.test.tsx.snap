// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGrid component... ...renders without errors 1`] = `ShallowWrapper {}`;
